lumigator/jobs/inference/definition.py

```diff
@@ -0,0 +1,84 @@
+from uuid import UUID
+
+from inference.schemas import (
+    DatasetConfig,
+    HuggingFacePipelineConfig,
+    InferenceJobConfig,
+    InferenceServerConfig,
+    JobConfig,
+    SamplingParameters,
+)
+from lumigator_schemas.jobs import JobCreate, JobType
+
+from backend.services.exceptions.job_exceptions import JobValidationError
+from backend.services.job_interface import JobDefinition
+
+# Served models
+OAI_API_URL: str = "https://api.openai.com/v1"
+MISTRAL_API_URL: str = "https://api.mistral.ai/v1"
+DEEPSEEK_API_URL: str = "https://api.deepseek.com/v1"
+
+DEFAULT_SUMMARIZER_PROMPT: str = "You are a helpful assistant, expert in text summarization. For every prompt you receive, provide a summary of its contents in at most two sentences."  # noqa: E501
+
+
+class JobInterfaceInference(JobDefinition):
+    def generate_config(self, request: JobCreate, record_id: UUID, dataset_path: str, storage_path: str):
+        # TODO Move to a custom validator in the schema
+        if request.job_config.task_definition.task.value == "text-generation" and not request.job_config.system_prompt:
+            raise JobValidationError("System prompt is required for text generation tasks.") from None
+        job_config = InferenceJobConfig(
+            name=f"{request.name}/{record_id}",
+            dataset=DatasetConfig(path=dataset_path),
+            job=JobConfig(
+                max_samples=request.max_samples,
+                storage_path=storage_path,
+                # TODO Should be unnecessary, check
+                output_field=request.job_config.output_field or "predictions",
+            ),
+        )
+        if request.job_config.provider == "hf":
+            # Custom logic: if provider is hf, we run the hf model inside the ray job
+            job_config.hf_pipeline = HuggingFacePipelineConfig(
+                model_name_or_path=request.job_config.model,
+                task=request.job_config.task_definition.task,
+                accelerator=request.job_config.accelerator,
+                revision=request.job_config.revision,
+                use_fast=request.job_config.use_fast,
+                trust_remote_code=request.job_config.trust_remote_code,
+                torch_dtype=request.job_config.torch_dtype,
+                max_new_tokens=500,
```

Comment:
This should be gone, maybe a missing merge?

Reply:
Nah, as I mentioned, moved the code, so git doesn't realize the situation :-/
Double checking 👍

Reply:
Manually checked as mentioned. Should be ok now.